src/webhooks/providers/bit2me/index.ts

```diff
@@ -0,0 +1,128 @@
+import { FastifyInstance } from 'fastify';
+import jwt from 'jsonwebtoken';
+
+import { StorageService } from '../../../services/storage.service';
+import { UsersService } from '../../../services/users.service';
+import { PaymentService } from '../../../services/payment.service';
+import { AppConfig } from '../../../config';
+import CacheService from '../../../services/cache.service';
+import { ObjectStorageService } from '../../../services/objectStorage.service';
+import { TiersService } from '../../../services/tiers.service';
+import { BadRequestError } from '../../../errors/Errors';
+import { InvoiceCompletedHandler } from '../../events/invoices/InvoiceCompletedHandler';
+import { DetermineLifetimeConditions } from '../../../core/users/DetermineLifetimeConditions';
+import { ObjectStorageWebhookHandler } from '../../events/ObjectStorageWebhookHandler';
+
+export interface CryptoWebhookDependencies {
+  storageService: StorageService;
+  cacheService: CacheService;
+  config: AppConfig;
+  objectStorageService: ObjectStorageService;
+  paymentService: PaymentService;
+  tiersService: TiersService;
+  usersService: UsersService;
+}
+
+export interface CryptoWebhookTokenPayload {
+  invoiceId: string;
+  provider: string;
+  customerId: string;
+}
+
+export interface Bit2MePaymentStatusCallback {
+  id: string;
+  foreignId: string;
+  cryptoAddress: {
+    currency: string;
+    address: string;
+  };
+  currencySent: {
+    currency: string;
+    amount: string;
+    remainingAmount: string;
+  };
+  currencyReceived: {
+    currency: string;
+  };
+  token: string;
+  transactions: any[];
+  fees: any[];
+  error: any[];
+  status: 'new' | 'pending' | 'confirming' | 'paid' | 'expired' | 'paid_after_expired';
+}
+
+export default function ({
+  storageService,
+  cacheService,
+  config,
+  objectStorageService,
+  paymentService,
+  tiersService,
+  usersService,
+}: CryptoWebhookDependencies) {
+  return async function (fastify: FastifyInstance) {
+    const decodeToken = async (token: string): Promise<CryptoWebhookTokenPayload> => {
+      return jwt.verify(token, config.JWT_SECRET) as CryptoWebhookTokenPayload;
+    };
+
+    fastify.post<{ Body: Bit2MePaymentStatusCallback }>('/webhook/crypto', async (req, rep) => {
```

Comment:
What about adding a try-catch to report the error to Bit2Me, or to log something at least?